R/zchunk_LB132.ag_an_For_Prices_USA_C_2005.R

```diff
@@ -38,43 +39,237 @@ module_aglu_LB132.ag_an_For_Prices_USA_C_2005_DISABLED <- function(command, ...)
+        mutate(V_USD = price * prod) %>%
+        left_join(FAO_an_items_PRODSTAT, by = "item") %>%
+        filter(!is.na(GCAM_commodity)) %>%
+        # Aggregate revenue and production by GCAM commodity
+        group_by(GCAM_commodity) %>%
+        summarize_at(vars(V_USD, prod), sum, na.rm = TRUE) %>%
+        ungroup() %>%
+        # Calculate production weighted average price for each GCAM commodity
+        mutate(Price_USDt = V_USD / prod) %>%
+        # Convert to model units
+        mutate(calPrice = round(Price_USDt * CONV_T_METRIC_SHORT * CONV_2004_1975_USD / CONV_T_KG, digits = DIGITS_CALPRICE)) %>%
+        select(GCAM_commodity, calPrice) ->
+        Price_an
+      } else {
+        # Animal products
+        FAO_USA_an_Prod_t_PRODSTAT %>%
+          select(-country.codes, -item.codes, -element, -element.codes) %>%
+          gather(year, prod, -countries, -item) %>%
+          # Calculate average production for each FAO animal product over price years
+          filter(year %in% MODEL_PRICE_YEARS) %>%
+          group_by(countries, item) %>%
+          summarise_at(vars(prod), mean, na.rm = TRUE) %>%
+          ungroup() %>%
+          filter(!is.na(prod)) %>%
+          # Match production and price for each FAO item to calculate revenue, avoid any missing value by inner_join
+          inner_join(Avg_price, by = c("countries", "item")) %>%
+          # Calculate revenue by commodity as production times price
+          mutate(V_USD = price * prod) %>%
+          left_join(FAO_an_items_PRODSTAT, by = "item") %>%
+          filter(!is.na(GCAM_commodity)) %>%
+          # Aggregate revenue and production by GCAM commodity
+          group_by(GCAM_commodity) %>%
+          summarize_at(vars(V_USD, prod), sum, na.rm = TRUE) %>%
+          ungroup() %>%
+          # Calculate production weighted average price for each GCAM commodity
+          mutate(Price_USDt = V_USD / prod) %>%
+          # Convert to model units, do not need to convert to short ton, drop CONV_T_METRIC_SHORT
+          mutate(calPrice = round(Price_USDt * CONV_2004_1975_USD / CONV_T_KG, digits = DIGITS_CALPRICE)) %>%
+          select(GCAM_commodity, calPrice) ->
+          Price_an
+      }
 
     # Produce outputs
-    # Temporary code below sends back empty data frames marked "don't test"
-    # Note that all precursor names (in `add_precursor`) must be in this chunk's inputs
-    # There's also a `same_precursors_as(x)` you can use
-    # If no precursors (very rare) don't call `add_precursor` at all
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+    Price_ag_an_For %>%
+      add_title("Prices for all GCAM AGLU commodities") %>%
+      add_units("1975$/kg and 1975$/m3") %>%
+      add_comments("Calculate unweighted average prices over calibration years by agricultural item and country.") %>%
```

Comment:
This comment and the next could be clearer. Maybe say something like "Calculate average prices over calibration years by GCAM commodity and region" for the first line. And, "Averages across years are unweighted; averages over item and country are weighted by production."